test/tests/smoke_test.rs

```diff
@@ -0,0 +1,43 @@
+// Licensed under the Apache-2.0 license
+
+use caliptra_builder::{ImageOptions, APP_WITH_UART, FMC_WITH_UART, ROM_WITH_UART};
+use caliptra_hw_model::{HwModel, InitParams};
+
+#[track_caller]
+fn assert_output_contains(haystack: &str, needle: &str) {
```

Comment:
Does this attribute report the error location in the firmware or the error location in the test flow ? Not clear on how this gets triggered.